EdgeTX/edgetx
# chore(cpn) - Upgrade Companion toolchain for Qt 5.15.2

Summary of changes:
- upgrade to miniz 3.0.2
- upgrade to QCustomplot 2.1.1
- upgrade to maxlibqt latest github and remove OpenTX implementation
- upgrade to Qt 5.15.2 (last to support 32 bit Win)
- eliminate qt depreciated warnings
- substitute linuxdeploy and linuxdeploy-plugin-qt for linuxdeployqt and new AppImage structure
- windeployqt run in debug mode as a workaround for a bug (Qt appears to have no intention of fixing)
- cmake use AUTOMOC and AUTOUIC for qt
- cmake new macros
- cmake housekeeping
- update github workflows
- update dfu-util and libusb sourcing to latest versions across oses

Notes:
1. repo build-edgetx has PR https://github.com/EdgeTX/build-edgetx/pull/16 to update Dockerfile
2. Qt 6 drops support for Win 32 bit and requires CXX 17 and has more depreciated and obsolete functions so another tranche of changes which will be done in another PR
3. Updating the compiler standard to CXX 17 generates lots of compiler warnings when compiling the libsims 
4. There are many non-Qt CXX 11 compiler warnings which need addressing some of which only show up based on compiler and/or Debug/Release mode
5. ~~Need to achieve consistency of dfu-util and libusb versions across OSes eg Win is running dfu-util v0.7 and libusb1.0 which is sourced from EdgeTX repo https://github.com/EdgeTX/edgetx/tree/main/companion/targets/windows. IMO should be latest versions unless compatibility issues and preferably sourced from devs repos.~~

Tested Ubuntu 20.04 modified build-edge Docker container, MSYS2 32 and 64 bit. Need some MacOS testers especially the building and running of the installer package.
